### ISSUE TITLE
Make it easier to enter 2FA code (magic links & formatting)

### ISSUE THREAD

**Issue by adriannelee** [cds-snc/notification-planning#603]
## Problem statement
Copying and pasting didn't seem easy for participants who used assistive devices (head pointer device) because it required a lot of precision. 

## Hypothesis
Making elements that are frequently copied and pasted (e.g. 2FA code) easy to copy and paste will save people some frustration. Eliminating the need to copy/paste a 2fa code will reduce user friction and increase accessibility.

We can't do buttons in emails. But we can make the target larger to select. But the only way to do this is by using headings (which isn't good practice)

Other services also have 2FA emails that this could benefit

Acceptance Criteria

- [ ] Put 2fa code on it's own line
- [ ] Add formatting with a grey bar to help make the 2fa stand out
- [ ] explore feasibility of using magic links
- [ ] ADR for magic links
- [ ] Security review 
Implementation of magic links so users can click a link and their 2fa code is entered for them
- [ ] API changes for magic links
- [ ] Admin changes for magic links
- [ ] Added templates for magic links
- [ ] Email 2fa template includes both the 2fa code and a magic link

Considerations
Magic links are only for email templates

**Comment by adriannelee** [comment]
Relevant finding from Forms team https://docs.google.com/presentation/d/1r8L2H-CEB5p6BEkU376xQAfPmOcQseDu/edit#slide=id.p9

**Comment by yaelberger-commits** [comment]
Add "Copy 2fa code" button to email that gets sent when a user signs in
Can reuse pattern/component we already use to copy template ID, for example.

**Comment by YedidaZalik** [comment]
Related to autocomplete

**Comment by yaelberger-commits** [comment]
Maybe we can use a magic link, to prove that they have access to their email, and the link contains the 2fa. Only for email

**Comment by andrewleith** [comment]
This is marked as low effort but I think that depends on what solution we go with here:
- styling the 2FA code to be easier to select using header markdown (low effort, but not recommended)
- styling the 2FA code to be easier to select using CSS (not low effort)
- using  magic link (not low effort)

@yaelberger-commits, thoughts?

**Comment by andrewleith** [comment]
- Threat modelling scheduled with SRE team
- Changing authentication mechanism is out of scope for now (we could revisit this if SRE thinks we have fundamental issues with our auth)
- As long as these new ideas dont add additional security risk, we will proceed with them
- Emails are sent to government emails only, which may increase our security stance

**Comment by YedidaZalik** [comment]
Threat modelling done
No new threats appeared to be introduced
Will to do ADR

**Comment by YedidaZalik** [comment]
Waiting on ADR threat model for magic link idea
So for now updating template as an improvement and Andrew hopes to work on today

**Comment by whabanks** [comment]
[ADR](https://github.com/cds-snc/notification-adr/pull/64) is ready for initial reviews.

**Comment by YedidaZalik** [comment]
Working on this today

**Comment by yaelberger-commits** [comment]
Small issue with formatting adding extra ** to the code in email previews. @andrewleith will look into removing the bold and keeping the grey bar
